test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -36,49 +36,80 @@
 #include "error_simulator.h"
 #include "timestamp_manager.h"
 
-session_simulator::session_simulator()
-    : _has_commit_ts(false), _ts_round_prepared(false), _ts_round_read(false), _txn_running(false),
-      _commit_ts(0), _durable_ts(0), _first_commit_ts(0), _prepare_ts(0), _read_ts(0)
-{
-}
+session_simulator::session_simulator() : _txn_running(false) {}
 
 int
 session_simulator::begin_transaction(const std::string &config)
 {
     /* Make sure that the transaction from this session isn't running. */
     assert(!_txn_running);
 
+    /* Reset timestamps */
+    _commit_ts = 0;
```

Comment:
You probably want to create a `reset` function that would call here and in the constructor too.

Reply:
Updated in [2e974b1](https://github.com/wiredtiger/wiredtiger/pull/8372/commits/2e974b18e64a8cde50905be6f7c41d9c0b0c3f92)